canvasapi/discussion_topic.py

```diff
@@ -18,6 +20,13 @@ def _parent_id(self):
             return self.course_id
         elif hasattr(self, "group_id"):
             return self.group_id
+        elif hasattr(self, "context_code"):
+            if re.match("course", self.context_code):
```

Comment:
If the intention here is to parse the string to figure out if it contains `"course"` (or later, `"group"`), that can be achieved without `re` with `"course" in self.context_code`.

If position is important, `self.context_code.startswith("course")` should also work.

Reply:
Wouldn't checking for "course" using `re` throw a `KeyError` if the context code isn't present? That's why I checked with `hasattr` first.

Reply:
Yeah checking with `hasattr` first is fine. I'm saying you don't need to use `re`:

```suggestion
            if self.context_code.startswith("course"):
```